src/number_theory/bernoulli_polynomials.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2021 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Ashvni Narayanan
+-/
+
+import number_theory.bernoulli
+
+/-!
+# Bernoulli polynomials
+
+The Bernoulli polynomials are an important tool obtained from Bernoulli numbers.
+
+## Mathematical overview
+
+The $n^ℐrm{th}$ Bernoulli polynomial is defined as
+$$ B_n(X) = ∑_{k = 0}^n {n χoose k} (-1)^k * B_k * X^{n - k} $$
+where $B_k$ is the $k^ℐrm{th}$ Bernoulli number. The Bernoulli polynomials are generating functions,
+$$ t * e^{tX} / (e^t - 1) = ∑_{n = 0}^{∞} B_n(X) * ¼{t^n}{n!} $$
+
+## Implementation detail
+
+The Bernoulli (negative) numbers are first defined, `bernoulli_neg`, in order to make proving
+theorems using Bernoulli polynomials easier.
+
+## Main theorems
+
+- `sum_bernoulli_poly`: The sum of the $k^\mathrm{th}$ Bernoulli polynomial with binomial
+  coefficients up to n is `(n + 1) * X^n`.
+- `exp_bernoulli_poly`: The Bernoulli polynomials act as generating functions for the exponential.
+-/
+
+noncomputable theory
+open_locale big_operators
+open_locale nat
+
+open nat finset
+
+/-- The Bernoulli polynomials are defined in terms of the negative Bernoulli numbers. -/
+def bernoulli_poly (n : ℕ) : polynomial ℚ :=
+  ∑ i in range (n + 1), polynomial.monomial (n - i) ((bernoulli i) * (choose n i))
+
+lemma bernoulli_poly_def (n : ℕ) : bernoulli_poly n =
+  ∑ i in range (n + 1), polynomial.monomial i ((bernoulli (n - i)) * (choose n i)) :=
+begin
+  rw [←sum_range_reflect, add_succ_sub_one, add_zero, bernoulli_poly],
+  apply sum_congr, {refl,},
+  rintros x hx,
+  rw mem_range_succ_iff at hx, rw [choose_symm hx, nat.sub_sub_self hx],
+end
+
+namespace bernoulli_poly
+
+/-
+### examples
+-/
+
+section examples
+
+@[simp] lemma bernoulli_poly_zero : bernoulli_poly 0 = 1 :=
+by { rw bernoulli_poly, simp, }
+
+@[simp] lemma bernoulli_poly_zero' (n : ℕ) : (bernoulli_poly n).eval 0 = bernoulli n :=
+begin
+  rw [bernoulli_poly, polynomial.eval_finset_sum], simp only [polynomial.eval_monomial],
+  rw sum_range_succ,
+  simp only [add_right_eq_self, mul_one, cast_one, nat.sub_self, choose_self, pow_zero],
+  apply sum_eq_zero, rintros x hx,
+  rw [zero_pow', mul_zero], apply mem_range_sub_ne_zero hx,
+end
+
+end examples
```

Comment:
Yes, this is true, and is easily provable. However, in fact, it follows from the general statement : 

B_n(1 - x) = (-1)^n*B_n(x)

This would take some effort, and I think can go in Todo for now.